### PR TITLE
Add a version check for bugsnag-android dependency

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Adds a version check for the bugsnag-android dependency, and disables writing to proguard files if the version is high enough that consumer proguard rules were already added to that artefact.

## Changeset

Adds a method which collects all the dependencies for all the project's configurations, and filters these to find the version number of bugsnag-android. Disables automatic running of the Proguard config task if the version is greater than 4.6.0.

## Tests

Manual test by changing the artefact version number in a sample project.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,11 @@
+
+## 3.2.8 (2018-06-21)
```
Comment (kattrali) [inline CHANGELOG.md]: Should be `TBD`?
